### ISSUE TITLE
DataTableColumns search not accurate

### ISSUE THREAD

**Issue by kyuohoh** [grommet/grommet#6876]
When using DataTableColumns, the search feature isn't properly filtering the options.
For example, when there is an option for "Carbon Emission" and I type "Car", I expect the option to be visible but the option is nowhere to be found.

<img width="297" alt="image" src="https://github.com/grommet/grommet/assets/99282684/d690c3c4-c7ea-4bdc-8d5c-0f7c89c1c0f7">

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version: 2.33.0
- Browser Name and version: Google Chrome
- Operating System and version (desktop or mobile): Desktop

**Comment by kyuohoh** [comment]
@jcfilben  - here's the codesandbox link: https://codesandbox.io/s/youthful-tristan-f4phv2?file=/src/DataTable.js

**Comment by jcfilben** [comment]
Thanks for sharing the example! Marking this as a bug

**Comment by kyuohoh** [comment]
@jcfilben - thanks. Just out of curiosity, when could I expect the bug fix?

**Comment by jcfilben** [comment]
Hey @kyuohoh I don't have any timeline for this at the moment but if you are eager to get this fixed definitely feel free to file a pull request, we rely a lot on open source contributions!

**Comment by MikeKingdom** [comment]
It's matching the column `property` attribute instead of the `label` attribute. I think DataTableColumns is matching in the wrong order according to this line (79):
```jsx
        nextFilteredOptions = options.filter((o) =>
          (o.property ?? o.label ?? o)?.toLowerCase().includes(lowerSearch),
        );
```

**Comment by kyuohoh** [comment]
Thanks @MikeKingdom  - I did see that the order of operation is incorrect. We could concatenate the `o.property` and `o.label` as a single string to match the searchTerm so that both property and label are used for filtering.

**Comment by MikeKingdom** [comment]
> Thanks @MikeKingdom - I did see that the order of operation is incorrect. We could concatenate the `o.property` and `o.label` as a single string to match the searchTerm so that both property and label are used for filtering.

A reasonable workaround. And definitely we should fix this soonish (and you're always welcome to submit a PR :) )

**Comment by kyuohoh** [comment]
@MikeKingdom @jcfilben  - Sorry but how could I get the permission to submit a PR?

`remote: Permission to grommet/grommet.git denied to kyuohoh.
fatal: unable to access 'https://github.com/grommet/grommet.git/': The requested URL returned error: 403`

**Comment by MikeKingdom** [comment]
> @MikeKingdom @jcfilben - Sorry but how could I get the permission to submit a PR?
> 
> `remote: Permission to grommet/grommet.git denied to kyuohoh. fatal: unable to access 'https://github.com/grommet/grommet.git/': The requested URL returned error: 403`

You need to do it via a fork. And my preferred solution would be to simply swap the order of the property and label in the filter:
```jsx
        nextFilteredOptions = options.filter((o) =>
          (o.label ?? o.property ?? o)?.toLowerCase().includes(lowerSearch),
        );
```

**Comment by kyuohoh** [comment]
@MikeKingdom  I'm okay with the above solution. Here's the PR: https://github.com/grommet/grommet/pull/6881

**Comment by jcfilben** [comment]
Resolved by #6881
